### PR TITLE
feat(msw): possible to overwrite value when creating an `object` with `msw` mock

### PR DESCRIPTION
**Pull request by soartec-lab**
## Status

**READY**

## Description

I responded to part of https://github.com/anymaniax/orval/issues/822

If there is only one response case and the type is `object`, the returned value can be overwritten externally by specifying `overrideResponse` as an argument.
In order to handle the case where it is an `array` of `object` such as `Pets`, after calculating the return value, we determine whether overwriting is possible based on whether `overrideResponse` is included in it.

## Todos

- [x] Tests
- [x] Documentation
- [x] Changelog Entry (unreleased)

## Steps to Test or Reproduce

1. Specify basic pet schema as input
2. execute `orval`

```
orval
```

3. generated mock function bellow: 

```typescript
export const getListPetsMock = (overrideResponse?: any) => (Array.from({
  length: faker.number.int({ min: 1, max: 10 }) }, (_, i) => i + 1).map(() => ({
    id: faker.number.int({min: undefined, max: undefined}),
    name: faker.word.sample(),
    tag: faker.helpers.arrayElement([faker.word.sample(), undefined]),
    ...overrideResponse
  }))
)

export const getShowPetByIdMock = (overrideResponse?: any) => ({
  id: faker.number.int({min: undefined, max: undefined}),
  name: faker.word.sample(),
  tag: faker.helpers.arrayElement([faker.word.sample(), undefined]), ...overrideResponse
})
```

4. usage

```typescript

import { getListPetsMock, getShowPetByIdMock } from 'gen/pets.msw'

const pets = getListPetsMock({name: 'orverride'})
console.log(pets)
// => [{ id: 645061935104000, name: "orverride", tag: undefined }, { id: 83169910980608, name: "orverride", tag: undefined }]
​​
const pet = getShowPetByIdMock({name: 'orverride'})
console.log(pet)
// => { id: 7272122785202176, ​name: "orverride", tag: undefined }
```

### CODE REVIEW

**Comment by melloware** [comment]
@Will-Mann-16 can you review since you recently did a lot of work in Mock?

File: packages/mock/src/msw/index.ts [hunk 1]
```diff
@@ -80,7 +82,7 @@ export const generateMSW = (
     implementation: {
       function:
         value && value !== 'undefined'
-          ? `export const ${functionName} = () => (${value})\n\n`
+          ? `export const ${functionName} = (${isResponseOverridable ? `overrideResponse?: any` : ''}) => (${value})\n\n`
```
Comment (Will-Mann-16) [inline packages/mock/src/msw/index.ts:85]: Some ESLint config setups would throw errors if you try to spread an undefined value, is it worth setting this to a default param of `{}`, empty object?

**Comment by soartec-lab** [comment]
@Will-Mann-16 
Hi, could you review this as well?

**Review by melloware — APPROVED** [review]

**Comment by melloware** [comment]
It looks good to me if @Will-Mann-16 has an issue we can re-open.

**Review by Will-Mann-16 — COMMENTED** [review]
This was the only thing I saw - if it's tested and works I'm happy. Otherwise might need a separate issue but would be an easy fix.

File: packages/mock/src/msw/index.ts (hunk 1 above)
Comment (soartec-lab) [reply packages/mock/src/msw/index.ts:85]: @Will-Mann-16 

Thank you for your comment. That's good. I didn't have that perspective. I'll fix it in another PR.

**Pull request merged**